src/com/rcv/Tabulator.java

```diff
@@ -675,13 +680,62 @@ private boolean hasContinuingCandidates(SortedMap<Integer, Set<String>> rankToCa
       } // end looping over the rankings within one ballot
     } // end looping over all ballots
 
+    // Take the tallies for this round for each precinct and merge them into the main map tracking
+    // the tallies by precinct.
+    for (String precinct : roundTallyByPrecinct.keySet()) {
+      // the set of round tallies that we've built up so far for this precinct
+      Map<Integer, Map<String, BigDecimal>> roundTalliesForPrecinct =
+          precinctRoundTallies.get(precinct);
+      roundTalliesForPrecinct.put(currentRound, roundTallyByPrecinct.get(precinct));
+    }
+
     return roundTally;
   }
 
+  // function: getNewTally
+  // purpose: create a new initialized tally with all continuing candidates
+  // returns: initialized tally
+  private Map<String, BigDecimal> getNewTally() {
+    Map<String, BigDecimal> tally = new HashMap<>();
+    // initialize tallies to 0 for all continuing candidates
+    for (String candidateID : candidateIDs) {
+      if (getCandidateStatus(candidateID) == CandidateStatus.CONTINUING) {
+        tally.put(candidateID, BigDecimal.ZERO);
+      }
+    }
+    return tally;
+  }
+
+  // function: incrementTally
+  // purpose: add a vote (or fractional share of a vote) to a tally
```

Comment:
Needs params documented.

Reply:
Actually, this is the only one of the three new methods I introduced that has any params.

Reply:
Ahh yeah, good point. I've been including all the fields (including return for void) but putting "N/A" in cases like this.